src/coreclr/src/vm/dllimportcallback.cpp

```diff
@@ -961,9 +961,16 @@ void UMEntryThunk::Terminate()
     CONTRACTL
     {
         NOTHROW;
+        MODE_ANY;
     }
     CONTRACTL_END;
 
+    if (GetObjectHandle())
+    {
+        DestroyLongWeakHandle(GetObjectHandle());
+        this->m_pObjectHandle = 0;
+    }
+
     m_code.Poison();
```

Comment:
It may be better to poinson the code first before freeing the handle - to make the diagnostic of the API misuse a tiny bit more reliable.